lib/process_win32.cpp

```diff
@@ -3,6 +3,32 @@
 
 #include "rtl_exec.h"
 
+struct ThreadInfo {
+    HANDLE pHandle;
+    std::vector<unsigned char> *buffer;
+};
+
+static DWORD WINAPI reader_thread(LPVOID lpParam)
+{
+    ThreadInfo *ti = reinterpret_cast<ThreadInfo*>(lpParam);
+
+    while (ti->pHandle != INVALID_HANDLE_VALUE) {
+        char buf[1024];
+        DWORD n;
+        if (ti->pHandle != INVALID_HANDLE_VALUE) {
+            if (!ReadFile(ti->pHandle, buf, sizeof(buf), &n, NULL)) {
+                break;
+            }
+            std::copy(buf, buf+n, std::back_inserter(*ti->buffer));
+        }
+    }
+    if (ti->pHandle != INVALID_HANDLE_VALUE) {
+        CloseHandle(ti->pHandle);
```

Comment:
Same about closing being the responsibility of the main thread.

Reply:
Fixed.